[test] Use %refactor-check-compiles in more places

hamishknight opened the pull request:
Add a hidden frontend flag `-warn-on-editor-placeholder` that allows `%refactor-check-compiles` to be more lenient with placeholders, and update a bunch of async refactoring tests to use `%refactor-check-compiles` in more cases.

Note the diff is best viewed without whitespace.

rdar://81830915

hamishknight commented:
@swift-ci please test

hamishknight commented:
@swift-ci please test Windows

bnbarham reviewed (approved):
Thanks for doing all those!

hamishknight commented on test/refactoring/ConvertAsync/convert_result.swift:
```diff
@@ -1,513 +1,533 @@
+  // ERROR-BLOCK-COMMENT-NEXT:   // h
+  // ERROR-BLOCK-COMMENT-NEXT:   print("after")
+  // ERROR-BLOCK-COMMENT-NEXT:   // i
+  // ERROR-BLOCK-COMMENT-NEXT:   {{ }}
```
It's because we include the trailing trivia up until the next token, but in this case the next token gets removed by the transform, leaving a stray newline. At the time, it didn't seem like a case that would come up too often, given it should only happen for trailing comments at the end of braces. Though looking at it again, it seems like it should be pretty straightforward to fix, filed rdar://82072147 to take care of it. Thanks for pointing it out!

hamishknight committed:
[Frontend] Add -warn-on-editor-placeholder

This hidden frontend option lets us be more lax
when type-checking in the presence of editor
placeholders by treating them as holes during
constraint solving.

hamishknight committed:
Don't error on placeholders for refactor-check-compiles

This allows us to test more cases that type-check
modulo placeholders.

hamishknight committed:
Have refactor-check-compiles pass down -I -sdk -target

Forward these flags to both swift-frontend and
swift-refactor.

hamishknight committed:
[test] Use %refactor-check-compiles in more places

Update a bunch of async refactoring tests to use
%refactor-check-compiles in more cases.

hamishknight commented:
@swift-ci please smoke test
